compiler/src/dotty/tools/dotc/ast/tpd.scala

```diff
@@ -918,7 +918,7 @@ object tpd extends Trees.Instance[Type] with TypedTreeInfo {
         Typed(tree, TypeTree(defn.AnyRefType))
       }
       else tree.ensureConforms(defn.ObjectType)
-      receiver.select(defn.Object_ne).appliedTo(nullLiteral)
+      receiver.select(defn.Object_ne).appliedTo(nullLiteral).withSpan(tree.span)
```

Comment:
This line triggers the crash. I find it better to add `withSpan` for other constructors too.